File: dashboard/src/app/model/countryModel.ts

import { IEntityBase }          from './IEntityBase';
import { RegionModel }          from './regionModel';
import { Status } from './enumModel';

export class CountryModel implements IEntityBase{
    id: number;   
    name: string;
    code: string;
    status: Status;

 
    regions: RegionModel[];   

    constructor(){
        this.id = 0;
        this.name = null;
        this.code = null;
        this.regions = null;
    }
}

File: server/medicore.layer/MediCore.Service/Model/CountryModel.cs
using dna.core.data.Infrastructure;
using dna.core.service.Models.Abstract;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;

namespace MediCore.Service.Model
{
    public class CountryModel : IModelBase
    {
        public int Id { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(100, ErrorMessage = "StringLength")]
        public string Name { get; set; }

        [Required(ErrorMessage = "Required")]
        [StringLength(5, ErrorMessage = "StringLength")]
        public string Code { get; set; }

        [Required]
        public Status Status { get; set; }

        public virtual ICollection<RegionModel> Regions { get; set; }
    }
}
